docs/en/sql-reference/functions/array-functions.md

```diff
@@ -2144,22 +2144,35 @@ Result:
 
 ## arrayROCAUC
 
-Calculates the Area Under the Curve (AUC), which is a concept in machine learning.
+Calculates the area under the receiver operating characteristic (ROC) curve.
+A ROC curve is created by plotting True Positive Rate (TPR) on the y-axis and False Positive Rate (FPR) on the x-axis across all thresholds.
+The resulting value ranges from 0 to 1, with a higher value indicating better model performance.
+The ROC AUC (also known as simply AUC) is a concept in machine learning.
 For more details, please see [here](https://developers.google.com/machine-learning/glossary#pr-auc-area-under-the-pr-curve), [here](https://developers.google.com/machine-learning/crash-course/classification/roc-and-auc#expandable-1) and [here](https://en.wikipedia.org/wiki/Receiver_operating_characteristic#Area_under_the_curve).
 
 **Syntax**
 
 ``` sql
-arrayROCAUC(arr_scores, arr_labels[, scale])
+arrayROCAUC(arr_scores, arr_labels[, scale, arr_partial_offsets])
```

Comment:
It makes sense to be able to use them independently. Currently, the user can pass:

```
arrayROCAUC(arr_scores, arr_labels)
arrayROCAUC(arr_scores, arr_labels, scale)
arrayROCAUC(arr_scores, arr_labels, scale, arr_partial_offsets)
```

According to the proposed doc change above, I have to update the code so that it also accepts:

```
arrayROCAUC(arr_scores, arr_labels, arr_partial_offsets)
```

Is that right? Since `scale` and `arr_partial_offsets` types are different, we can parse its type to identify which optional argument was passed. 

A following question I have is, should the following usage be accepted too? Or should this raise an error:

```
arrayROCAUC(arr_scores, arr_labels, arr_partial_offsets, scale)
```

Reply:
> According to the proposed doc change above, I have to update the code so that it also accepts:
> ``` SQL
> arrayROCAUC(arr_scores, arr_labels, arr_partial_offsets)
> ```
> Is that right?

Yes, if we can do this, why not.

> A following question I have is, should the following usage be accepted too? Or should this raise an error:

It should raise an error. We should have our arguments in the determined order.